### PR TITLE
Added large object support for objects larger than 1GB

### PR DESCRIPTION
**Pull request by DRichardt**
+ updated Swift Resource

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,6 +65,8 @@ Given a path specified by `from`, upload it to the Swift container. The path mus
 
 * `from`: *Required.* A regexp specifying the file to upload.
 
+* `SegmentContainer` : *Required* Name of the container where segments should be stored. Segments will be created for files larger than 1 GB.
```
Comment (databus23) [inline README.md]: Is this really required? That would mean the change is not backward compatible, which would make me sad.

File: pkg/resource/out.go [hunk 2]
```diff
@@ -67,10 +69,62 @@ func Out(request OutRequest, sourceDir string) (*OutResponse, error) {
...
+
+	stat, err := file.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("Can't stats of file %s: %s", fileSource, err)
+	}
+
+	var bytes int64
+	bytes = stat.Size()
+	var fi os.FileInfo
+	if bytes > 134217728 {
```
Comment (databus23) [inline pkg/resource/out.go]: Is this size boundary arbitrarily chosen?

File: pkg/resource/models.go [hunk 3]
```diff
@@ -49,3 +50,19 @@ type OutResponse struct {
 	Version  Version    `json:"version"`
 	Metadata []Metadata `json:"metadata"`
 }
+type Headers map[string]string
+
+type LargeObjectOpts struct {
```
Comment (databus23) [inline pkg/resource/models.go]: Is this struct used anywhere?

File: pkg/resource/models.go [hunk 4]
```diff
@@ -49,3 +50,19 @@ type OutResponse struct {
 	Version  Version    `json:"version"`
 	Metadata []Metadata `json:"metadata"`
 }
+type Headers map[string]string
```
Comment (databus23) [inline pkg/resource/models.go]: Is this type used anywhere?

File: pkg/resource/out.go [hunk 5]
```diff
@@ -67,10 +69,62 @@ func Out(request OutRequest, sourceDir string) (*OutResponse, error) {
...
+
+		opts := swift.LargeObjectOpts{
+			Container:        rsc.Container,
+			ObjectName:       filename,
+			ContentType:      http.DetectContentType(fileHeader),
+			ChunkSize:        1073741824,
+			MinChunkSize:     1073741824,
+			SegmentContainer: request.Params.SegmentContainer,
+		}
+		//_, err := client.ObjectCreate(rsc.Container, filename, true, "", "", nil)
```
Comment (databus23) [inline pkg/resource/out.go]: To reduce clutter I would delete this unused code

File: pkg/resource/models.go (hunk 4 above)
Comment (DRichardt) [reply pkg/resource/models.go]: in line 63

File: pkg/resource/models.go (hunk 3 above)
Comment (DRichardt) [reply pkg/resource/models.go]: in the out.go

File: pkg/resource/out.go (hunk 5 above)
Comment (DRichardt) [reply pkg/resource/out.go]: OK

File: pkg/resource/out.go (hunk 2 above)
Comment (DRichardt) [reply pkg/resource/out.go]: It's proposed by a swift specialist ( larger 1GB )

File: README.md (hunk 1 above)
Comment (DRichardt) [reply README.md]: For uploading segments we need a segment container. A second option would be that default is to append _segments to the default container (like in swift cli client ) . SegmentContainer would be optional in this case.

File: pkg/resource/models.go (hunk 3 above)
Comment (databus23) [reply pkg/resource/models.go]: In `out.go` I can only find `swift.LargeObjectOpts` which is a struct that is defined in the swift client package, no?

File: README.md (hunk 1 above)
Comment (databus23) [reply README.md]: My point is that if I only upload small files i never need to worry about a segment container. So the `SegmentContainer` parameter is not really required. It is only required if you have a large uploads or specifically want to segment your uploads.

File: pkg/resource/out.go (hunk 2 above)
Comment (databus23) [reply pkg/resource/out.go]: In stead of having a hidden magical value that triggers upload segmentation maybe the user should have more control when and if segmentation kicks in.

File: README.md [hunk 6]
```diff
@@ -65,6 +65,9 @@ Given a path specified by `from`, upload it to the Swift container. The path mus
 
 * `from`: *Required.* A regexp specifying the file to upload.
 
+* `SegmentContainer` : *Optional* Name of the container where segments should be stored. Segments will be created for files larger than 1 GB. Default: (Container_segments)
```
Comment (databus23) [inline README.md]: Up until now the parameters where defined in snake case. So for consistency this should be changed to `segment_container` and `segment_size`. Also maybe the default value should be written as (Default: `$container`_segments)

Comment (DRichardt) [reply README.md]: should be done

**Pull request merged**